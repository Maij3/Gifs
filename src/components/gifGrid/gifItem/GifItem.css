.flex-item {
    margin-bottom: 20px;
    border: 1px solid #eee;
    width: 80%;
}
@media only screen and (max-width: 767px){
    .flex-item {
	border-radius: unset;
    }
}
