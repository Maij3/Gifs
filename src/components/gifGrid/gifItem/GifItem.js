import PropTypes from 'prop-types';
import { Box } from "@chakra-ui/react";
import './GifItem.css'
const GifItem = ({ url }) => {
  return (
    <Box width="33.333%" display='flex' alignItems='center' mt='20px'>
      <img
        className='flex-item'
        src={url}
        alt=""
      />
    </Box>
  );
};

GifItem.propTypes = {
    url : PropTypes.string.isRequired,
};

export default GifItem;
