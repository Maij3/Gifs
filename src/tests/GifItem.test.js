import React from "react";
import { render, screen } from "@testing-library/react";
import GifItem from "../components/gifGrid/gifItem/GifItem";
import {GridItem} from "@chakra-ui/react";
describe("Pruebas en <GifItem />", () => {
  const url = "https://jaime.com/";
  it("Debe Hacer Match con el Snapshot", () => {
    const { container } = render(<GifItem url={url} />);
    expect(container).toMatchSnapshot();
  });
  it("Debe Mostrar el Url Indicicado", () => {
    render(<GifItem url={url} />);
    expect(screen.getByRole("img").src).toBe(url);
  });
/*  it('Debe de Mostrar el Titulo en el Componente', () => {*/
	/*render( <GifItem url={url} /> );*/
	/*expect( screen.getByText( title ) ).toBeTruthy();*/
  /*})*/
});
