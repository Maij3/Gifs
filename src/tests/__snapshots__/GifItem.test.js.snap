// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe Hacer Match con el Snapshot 1`] = `
<div>
  <div
    class="css-ydo0x6"
  >
    <img
      alt=""
      class="flex-item"
      src="https://jaime.com/"
    />
  </div>
</div>
`;
